pinterest/teletraan
# Add an auth filter for PinDeploy pipeline

This PR implements an auth filter that will replace the principal if both of the conditions are met
1. Request is sent from a list of spiffe ids
2. Request contains a special header.
3. Because the principal name has a SPIFFE prefix, it's more likely to exceed the DB column limit. Introduced a method to trim the string to avoid updating the DB. 

## Test and validations
test coverage and some manual tests
1. Start the service with updated configuration
2. Craft a request with the spiffe ID and special header
4. Verify in the debugger that the principal is replaced.
